freyja_examples/src/temporal_provider.cpp

```diff
@@ -1,152 +1,91 @@
-TrajRef getCircleReference( const ros::Duration &cur_time, const int agg_level)
-{
-  // A is amplitude (radius); w angular rate such that 2pi/w = (seconds for one rev)
-  float A = 0.5;
-  float w = 0.5;
-
-  // Set A and w based on agg_level
-  switch(agg_level) {
-    case 1 :
-      break;
-    case 2 :
-      A = 0.5;
-      w = 1;
-      break;
-    case 3 :
-      A = 1;
-      w = 3;
-      break;
-    default :
-      ROS_WARN_STREAM("Circle aggression " << agg_level << " not supported, defaulting to agg_level 1");
-  }
-
-  float t = cur_time.toSec();
-
-  // Create reference state
-  TrajRef ref_state;
-  ref_state.header.stamp = ros::Time::now();
-
-  ref_state.pn = A*std::sin( w*t );
-  ref_state.pe = A*std::cos( w*t );
-  ref_state.pd = -1.0;
-  
-  ref_state.vn = A*w*std::cos( w*t );
-  ref_state.ve = -A*w*std::sin( w*t );
-  ref_state.vd = 0.0;
-
-  ref_state.yaw = 0.0;
-
-  // set an, ae, ad to second derivatives if needed for FF..
-  return ref_state;
-}
+rclcpp::Time init_time;
 
-TrajRef getDefaultReference( const ros::Duration &cur_time )
+class Temporal_Traj : public rclcpp::Node
 {
-  TrajRef ref_state;
-  ref_state.pn = 0.5;
-  ref_state.pe = 0.5;
-  ref_state.pd = -1.0;
-  ref_state.vn = 0.0;
-  ref_state.ve = 0.0;
-  ref_state.vd = 0.0;
-  ref_state.yaw = DEG2RAD(0.0);
-  ref_state.an = 0.0;
-  ref_state.ae = 0.0;
-  ref_state.ad = 0.0;
-  ref_state.header.stamp = ros::Time::now();
-  return ref_state;
-}
-
-int main( int argc, char** argv )
+	public:
+		TrajRef()
+		: Node(ROS_NODE_NAME)
+		{
+			this->declare_parameter("traj_type", traj_type);
+			this->get_parameter("traj_type", traj_type);
+
+			this->declare_parameter("agg_level", agg_level);
+  		this->get_parameter("agg_level", agg_level);
+
+			traj_pub = this->create_publisher<TrajRef>("/reference_state", 1);
+
+			time_reset_sub = this->create_subscription<std_msgs::msg::UInt8>(
+			"/reset_trajectory_time", 1, std::bind(&Temporal_Traj::timer_reset_cb, this, _1));
+
+			traj_update_timer = this->create_wall_timer(
+      1/UPDATE_RATE, std::bind(&SbusComm::traj_update_cb, this));
+
+		}
+
+	void timer_reset_cb( const std_msgs::msg::UInt8::SharedPtr msg )
+	{
+		if (msg -> data == 1)
+		{
+			init_time = now();
+			RCLCPP_WARN(this->get_logger(), "%s: Time reset requested!", ROS_NODE_NAME);
+		}
+	}
+
+	void traj_update_cb()
+	{
+		if (traj_type == "hover")
+		{
+			ref_state = getHoverReference(now() - init_time);
+		}
+		else if (traj_type == "circle")
+		{
+			ref_state = getCircleReference(now() - init_time, agg_level);
+		}
+		else if (traj_type == "lemiscate")
+		{
+			ref_state = getLemiscateReference(now() - init_time, agg_level);
+		}
+		else
+		{
+			RCLCPP_ERROR(this->get_logger(), "Invalid trajectory parameters given.")
+		}
+
+		traj_pub->publish(ref_state);
+	}
+
+	rclcpp::Publisher<TrajRef>::SharedPtr traj_pub;
+	rclcpp::Subscription<std_msgs::msg::UInt8>::SharedPtr time_reset_sub;
+	rclcpp::TimerBase::SharedPtr traj_update_timer;
+};
```

Comment:
I'd recommend formatting the class structure such that private members (as well declarations) appear early, and public member functions can be defined outside the class. Checkout the style in controller or state nodes.